ansible_base/rbac/api/serializers.py

```diff
@@ -242,6 +242,13 @@ def create(self, validated_data):
         # Resolve object
         obj = self.get_object_from_data(validated_data, rd, requesting_user)
 
+        # model-level callback to further validate the assignment
+        # can be optionally implemented by the model
+        if getattr(obj, 'validate_role_assignment', None):
+            errmsg = obj.validate_role_assignment(actor, rd)
+            if errmsg:
+                raise PermissionDenied(errmsg)
```

Comment:
The [prior implementation](https://github.com/ansible/awx/blob/fea9e0271cb1dbc7da60bce65756eb6c2092c694/awx/api/views/__init__.py#L4381-L4388) you are mimicking raised a 400, not a 403. I would favor the 400.